### PR TITLE
[FLINK-32884] [flink-clients] Sending messageHeaders with decorated customHeaders for PyFlink client

### PR DESCRIPTION
**Pull request by elkhand**
[FLINK-32884](https://issues.apache.org/jira/browse/FLINK-32884) PyFlink remote execution should support URLs with paths and https scheme

## What is the purpose of the change

This PR handles including customHeaders in messageHeaders when a Python client is sending requests. This will allow sending auth tokens with the request.

## Brief change log

- send messageHeaders which are decorated with customHeaders

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

This change is already covered by existing tests, such as `flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java#testRESTManualConfigurationOverride`

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper:no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by elkhand** [comment]
@flinkbot run azure

File: flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java [hunk 1]
```diff
@@ -909,6 +910,17 @@ void testSubmitJobAndWaitForExecutionResult() throws Exception {
         final AtomicBoolean firstSubmitRequestFailed = new AtomicBoolean(false);
         failHttpRequest =
                 (messageHeaders, messageParameters, requestBody) -> {
+                    if (messageHeaders instanceof CustomHeadersDecorator) {
```
Comment (tweise) [inline flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java:913]: This is a bit hard to follow. Shouldn't we expect to find `CustomHeadersDecorator` and throw an error otherwise? If not then the bug you are trying to fix could be re-introduced w/o the test breaking?

Comment (elkhand) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java:913]: We are interested in `messageHeaders` itself, and it is double wrapped in `CustomHeadersDecorator(new UrlPrefixDecorator(messageHeaders))`. The test was checking for instance type of `messageHeaders`, which this code is trying to do.
`RestClusterClient.sendRequest()` function internally calls `RestClient.sendRequest()` function which accesses messageHeaders and customHeaders to construct httpRequest, and it is not accessed anywhere else. That part in RestClient is already updated to work with decorated messageHeaders:
https://github.com/apache/flink/pull/22816/files#diff-d6cdab55fefade687bd62fb086b6fa85aa4f8c91a745e8ec406fb8ac8982b9e1R440 

https://github.com/apache/flink/pull/22556/files#diff-d6cdab55fefade687bd62fb086b6fa85aa4f8c91a745e8ec406fb8ac8982b9e1R358

Comment (tweise) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java:913]: My question is why the unwrapping is conditional? Isn't it an error if `messageHeaders` is not double wrapped? If so, then the test should check that.

Comment (elkhand) [reply flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java]: I see your point. I removed all checks, as now we will always wrap messageHeaders into `CustomHeadersDecorator(new UrlPrefixDecorator(messageHeaders))` before sending the request, thus we do not need those condition checks anymore.
